batch/sql/estimated-current.sql

```diff
@@ -1049,7 +1050,10 @@ BEGIN
                jobs.job_id = t.job_id
           SET jobs.state = IF(COALESCE(t.n_pending_parents, 0) = 0, 'Ready', 'Pending'),
               jobs.n_pending_parents = COALESCE(t.n_pending_parents, 0),
-              jobs.cancelled = IF(COALESCE(t.n_succeeded, 0) = COALESCE(t.n_parents - t.n_pending_parents, 0), jobs.cancelled, 1)
+              jobs.cancelled = IF(jobs.run_condition = 'all',
+                                  IF(COALESCE(t.n_succeeded, 0) = COALESCE(t.n_parents - t.n_pending_parents, 0), jobs.cancelled, 1),
+                                  IF(jobs.run_condition = 'always', 0,
+                                     (COALESCE(t.n_parents, 0) = 0) | COALESCE(t.n_succeeded, 0) > 0)  # run_condition = 'any'
```

Comment:
Nit: Just for ease of readability, can we use a [CASE expression](https://www.w3schools.com/sql/func_mysql_case.asp) instead of the nested `IF`? Feels like it should map better to the all / any / always enum.